File: 704二分查找/Java/Sol.java
class Sol {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        int mid;
        while (left <= right) {
           mid = (left + right) / 2;
           if(nums[mid] < target) {
               left = mid + 1;
           } else if(nums[mid] > target) {
               right = mid -1;
           } else {
               return mid;
           }
        } 
        return -1;
    }
    public static void main(String[] args) {
        int [] nums = new int[]{-1,0,3,5,9,12};
        int target = 2;
        Sol s = new Sol();
        int res = s.search(nums,target);
        System.out.println(res);
    }
}

File: 704二分查找/CPP/Sol.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        int mid = (left + right) / 2;
        while (left <= right) {
        // 直接判断中间时间开销较大
        //    if (nums[mid] == target) {
        //        return mid;
        //    } else if (nums[mid] < target) {
        //        left = mid + 1;
        //    } else {
        //        right = mid - 1;
        //    }
            if (num[mid] > target) {
                right = mid - 1;
            } else if (num[mid] < target) {
                left = mid + 1;
            } else {
                return mid;
            }
            mid = (left + right) / 2;
        }
        return -1;
    }
};

int main() {
    vector<int> nums = {-1,0,3,5,9,12};
    int target = 9;
    Solution sol;
    int res = sol.search(nums,target);
    cout << res << endl;
}
